### ISSUE TITLE
Zero gradient

### ISSUE THREAD

**Issue by elbamos** [caogang/wgan-gp#16]
I'm trying to implement the gradient-penalty approach using this code, but this code block: 
```
    gradients = autograd.grad(outputs=disc_interpolates, inputs=interpolates,
                              grad_outputs=torch.ones(disc_interpolates.size()).cuda(gpu) if use_cuda else torch.ones(
                                  disc_interpolates.size()),
                              create_graph=True, retain_graph=True, only_inputs=True)[0]
```
always returns gradients of 0.  

Would this be caused by my using `ConvTranspose2d` units?

**Comment by caogang** [comment]
Where are you putting `ConvTranspose2d` in, Discriminator or Generator? And do you use `Dilated` feature(Because current double backward does not support `dilated` feature)? Do you use the latest master branch?

**Comment by elbamos** [comment]
Generator, without dilation. But you're right, if the modules are on the generator, that can't be what's causing this.

**Comment by elbamos** [comment]
I traced it to an `InstanceNorm2d` layer... Anyway it wasn't an issue in your code, thanks again!
